Open Colors, add author, update deps (#79)
excalidraw/excalidraw-blog: content/blog/open-colors/index.md

---
title: Open Colors
date: 2020-11-09
author: Lipis
link: https://twitter.com/lipis
---

> 

Reviewer: ```suggestion

(Part of a series where we pick apart various libraries and projects we use on Excalidraw.)
```

Reviewer: Best if we also come up with a title prefix as discussed.

Author: I added that as a note in frontmatter...

One of the qualities of Excalidraw is its simplicity. Even though we have the option to use any color of the [spectrum](https://www.google.com/search?q=%23c0ffee&hl=en), we have decided to limit our palette to a curated set of 15 colors in three different shades.

<!-- end -->

> The [Open Colors](https://yeun.github.io/open-color/) color scheme has a total of 13 colors with 10 different brightnesses (0-9). For the canvas background we chose the lightest of the values (0), for the strokes the darkest one (9) and for the background fill of the elements slightly lighter than the stroke, we went with the 6th (or the 7th, dependas on how you count them).

Reviewer: ```suggestion
The [Open Colors](https://yeun.github.io/open-color/) color scheme has a total of 13 colors with 10 different brightnesses (0-9). For the canvas background we chose the lightest of the values (0), for the strokes the darkest one (9), and for the element background fill we went with the 6th (or the 7th, depending on how you count them).
```

> The [Open Colors](https://yeun.github.io/open-color/) color scheme has a total of 13 colors with 10 different brightnesses (0-9). For the canvas background we chose the lightest of the values (0), for the strokes the darkest one (9) and for the background fill of the elements slightly lighter than the stroke, we went with the 6th (or the 7th, dependas on how you count them).

Reviewer: ```suggestion
The [Open Colors](https://yeun.github.io/open-color/) color scheme has a total of 13 colors with 10 different brightness levels (0-9). For the canvas background we chose the lightest of the values (0), for the strokes the darkest one (9) and for the background fill of the elements slightly lighter than the stroke, we went with the 6th (or the 7th, dependas on how you count them).
```

---
title: Open Colors
date: 2020-10-29
author: Lipis
link: https://twitter.com/lipis
---

> The biggest attribute of Excalidraw it's the simplicity. So even though we have the option to insert any color (using the [HEX](https://www.google.com/search?q=%23c0ffee&hl=en) value) we have just the _right amount_ (15) of the default colors, in three different shades.

Reviewer: ```suggestion
One of the qualities of Excalidraw is its simplicity. Even though we have the option to use any color of the [spectrum](https://www.google.com/search?q=%23c0ffee&hl=en), we have decided to limit our palette to a curated set of 15 colors in three different shades.
```

<!-- end -->

![Color pickers](color-pickers.png)

> I remember it was my very first [pull request](https://github.com/excalidraw/excalidraw/pull/378) to Excalidraw to add the [Open Colors](https://yeun.github.io/open-color/) open source color scheme and it was accepted the same day! I was super excited to say the least, because I knew this color scheme for a while now, but never found a great project to actually use it. This was it.

Reviewer: ```suggestion
I remember it was my very first [pull request](https://github.com/excalidraw/excalidraw/pull/378) to Excalidraw to add the [Open Colors](https://yeun.github.io/open-color/) color scheme. I've known about this color scheme for a while now, but never found a great project to actually use it on. Excalidraw was a perfect fit.
```

> They say a picture is worth a thousand words! I don't know what makes this color scheme so great, but what I know is that anything in Excalidraw is making it even better. (Click on the image below to open it in Excalidraw.)

Reviewer: ```suggestion
They say a picture is worth a thousand words. I don't know what makes this color scheme so great, but something about Excalidraw is making it look even better. (Click on the image below to open it in Excalidraw.)
```

https://excalidraw.com/#json=5120999011909632,Y57VloPaA1LSKT4-1NTgNA

> Eventually we loved these colors so much, that we installed their [npm package](https://www.npmjs.com/package/open-color) and we are now using it everywhere. If you spot a color that is not an open color, sumbit a PR :)

Reviewer: ```suggestion
Eventually we decided to install their [npm package](https://www.npmjs.com/package/open-color) so we don't have to hand-code every value. We use it both in [CSS](https://github.com/excalidraw/excalidraw/blob/master/src/css/_variables.scss), and [TypeScript](https://github.com/excalidraw/excalidraw/blob/master/src/colors.ts).
```

> 
> Here is how we are actually using them in [colors.ts](https://github.com/excalidraw/excalidraw/blob/master/src/colors.ts) in our codebase.

Reviewer: ```suggestion
```

> Open Colors are coming with MIT License, which means you can use it anywhere you like without restrictions.

Reviewer: ```suggestion
Open Colors come with MIT License, which means you can use it anywhere you like without restrictions.
```

---

> **P.S.** We owe a huge thanks to the [Open Colors](https://yeun.github.io/open-color/) creator, as we trully believe that part of Exaclidraw's success is these colors as well :)

Reviewer: Thanks [@yeun](https://github.com/yeun) and the contributors for creating and maintaining the project!
